src/ToDoContext.tsx

```diff
@@ -0,0 +1,181 @@
+/* eslint-disable no-param-reassign */
+// eslint-disable-next-line
+import _ from 'lodash';
+import React, { useEffect, useReducer } from 'react';
+import { ToDo } from './types/ToDo';
+
+export enum ACTIONS {
+  ADD = 'addPost',
+  REMOVE = 'removePost',
+  UPDATE = 'updatePost',
+  TOGGLE = 'completed',
+  SORT = 'sortBy',
+  TOGGLE_ALL = 'TOGGLE_ALL',
+  CLEAR = 'removeComplited',
+
+}
+
+type Action = { type: ACTIONS.ADD, payload: string }
+| { type: ACTIONS.UPDATE, payload: ToDo }
+| { type: ACTIONS.REMOVE, payload: ToDo }
+| { type: ACTIONS.TOGGLE, payload: number }
+| { type: ACTIONS.SORT, payload: string }
+| { type: ACTIONS.TOGGLE_ALL, payload: boolean }
+| { type: ACTIONS.CLEAR };
+
+interface State {
+  list: ToDo[];
+  sortBy: string;
+}
+
+export enum FILTER {
+  ALL = 'ALL',
+  ACTIVE = 'ACTIVE',
+  COMPLITED = 'COMPLITED',
+}
+
+function newToDo(name: string): ToDo {
+  return { id: Date.now(), title: name, completed: false };
+}
+
+function toggle(list: ToDo[], id: number): ToDo[] {
+  const copyList = _.cloneDeep(list);
```

Comment:
```suggestion
  return cloneDeep(list).map((...
```

Reply:
because you should also change it in here
<img width="394" alt="image" src="https://github.com/mate-academy/react_todo-app/assets/70220667/fd93b4f3-e8f5-45de-98b9-097a827a0c36">